numpy/linalg/linalg.py

```diff
@@ -1672,19 +1643,17 @@ def pinv(a, rcond=1e-15 ):
 
     """
     a, wrap = _makearray(a)
-    if _isEmpty2d(a):
-        res = empty(a.shape[:-2] + (a.shape[-1], a.shape[-2]), dtype=a.dtype)
-        return wrap(res)
     a = a.conjugate()
     u, s, vt = svd(a, 0)
     m = u.shape[0]
     n = vt.shape[1]
-    cutoff = rcond*maximum.reduce(s)
```

Comment:
`maximum.reduce` fails on empty arrays with `ValueError: zero-size array to reduction operation maximum which has no identity`. Shouldn't this just return `-inf` for arrays of dtype `inexact`?